File: app/src/main/java/com/arturogutierrez/openticator/domain/account/add/di/AddAccountComponent.java
package com.arturogutierrez.openticator.domain.account.add.di;

import com.arturogutierrez.openticator.di.PerActivity;
import com.arturogutierrez.openticator.di.component.ActivityComponent;
import com.arturogutierrez.openticator.di.component.ApplicationComponent;
import com.arturogutierrez.openticator.di.module.ActivityModule;
import com.arturogutierrez.openticator.domain.account.add.view.AddAccountFormFragment;
import dagger.Component;

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = {
    ActivityModule.class, AddAccountModule.class
})
public interface AddAccountComponent extends ActivityComponent {
  void inject(AddAccountFormFragment addAccountFormFragment);
}

File: app/src/main/kotlin/com/arturogutierrez/openticator/domain/account/add/di/AddAccountComponent.kt
package com.arturogutierrez.openticator.domain.account.add.di

import com.arturogutierrez.openticator.di.PerActivity
import com.arturogutierrez.openticator.di.component.ActivityComponent
import com.arturogutierrez.openticator.di.component.ApplicationComponent
import com.arturogutierrez.openticator.di.module.ActivityModule
import com.arturogutierrez.openticator.domain.account.add.view.AddAccountFormFragment
import com.arturogutierrez.openticator.domain.account.add.view.AddAccountFromCameraFragment
import dagger.Component

@PerActivity
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(ActivityModule::class, AddAccountModule::class))
interface AddAccountComponent : ActivityComponent {

  fun inject(addAccountFormFragment: AddAccountFormFragment)

  fun inject(addAccountFromCameraFragment: AddAccountFromCameraFragment)
}
